### ISSUE TITLE
Error in rule barcode_report when running harpy align (step 11 of 14)

### ISSUE THREAD

**Issue by prmunn** [pdimens/harpy#143]
### Describe the bug

Not sure from the log file what caused this or how to run it again manually. Log file attached, so hopefully that give you enough info to figure out what's going on.

### Is this the latest version of Harpy?

- [X] yes
- [ ] no

### Harpy error log

```console
[Mon Oct  7 16:59:31 2024]
localrule barcode_report:
    input: Align/bwa/reports/data/bxstats/C3-flyS3-10477404-22NLWYLT3-L2_trimmed.bxstats.gz
    output: Align/bwa/reports/barcodes.summary.html
    jobid: 13
    reason: Missing output files: Align/bwa/reports/barcodes.summary.html; Input files updated by another job: Align/bwa/reports/data/bxstats/C3-flyS3-10477404-22NLWYLT3-L2_trimmed.bxstats.gz
    resources: tmpdir=/tmp

Activating conda environment: .snakemake/conda/52de4b1d31cff955b1d021a7fa1b5edf_
RuleException:
CalledProcessError in file /local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly/Align/bwa/workflow/align_bwa.smk, line 241:
Command 'source /home/prm88/mambaforge-pypy3/bin/activate '/local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly/.snakemake/conda/52de4b1d31cff955b1d021a7fa1b5edf_'; set -euo pipefail;  Rscript --vanilla -e 'rmarkdown::render("/local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly/.snakemake/scripts/tmperrnh_ha.align_bxstats.Rmd", output_file="/local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly/Align/bwa/reports/barcodes.summary.html", quiet=TRUE, knit_root_dir = "/local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly", params = list(rmd="/local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly/.snakemake/scripts/tmperrnh_ha.align_bxstats.Rmd"))'' returned non-zero exit status 1.
[Mon Oct  7 17:00:04 2024]
Error in rule barcode_report:
    jobid: 13
    input: Align/bwa/reports/data/bxstats/C3-flyS3-10477404-22NLWYLT3-L2_trimmed.bxstats.gz
    output: Align/bwa/reports/barcodes.summary.html
    conda-env: /local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly/.snakemake/conda/52de4b1d31cff955b1d021a7fa1b5edf_

[Mon Oct  7 17:00:05 2024]
Finished job 10.
11 of 14 steps (79%) done
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2024-10-07T154820.749839.snakemake.log
```

**Comment by pdimens** [comment]
Does this error still occur when using `--conda` ?

**Comment by prmunn** [comment]
Yes. 

Here's the command I was using: harpy align bwa -t 20 --conda -g /local/workdir/genomes/Drosophila_melanogaster/dm6/ENSEMBL/Drosophila_melanogaster.BDGP6.32.dna.toplevel.fa 2024_10_02_trimmed_fastqs

**Comment by pdimens** [comment]
That's unexpected. What happens if you delete the `.snakemake/` folder and try that command again?

**Comment by prmunn** [comment]
I'll give that a try...

**Comment by prmunn** [comment]
Nope - same error:

[Tue Oct  8 14:54:59 2024]
localrule barcode_report:
    input: Align/bwa/reports/data/bxstats/C3-flyS3-10477404-22NLWYLT3-L2_trimmed.bxstats.gz
    output: Align/bwa/reports/barcodes.summary.html
    jobid: 13
    reason: Missing output files: Align/bwa/reports/barcodes.summary.html
    resources: tmpdir=/tmp

Activating conda environment: .snakemake/conda/52de4b1d31cff955b1d021a7fa1b5edf_
Activating conda environment: .snakemake/conda/52de4b1d31cff955b1d021a7fa1b5edf_
RuleException:
CalledProcessError in file /local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly/Align/bwa/workflow/align_bwa.smk, line 241:
Command 'source /home/prm88/mambaforge-pypy3/bin/activate '/local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly/.snakemake/conda/52de4b1d31cff955b1d021a7fa1b5edf_'; set -euo pipefail;  Rscript --vanilla -e 'rmarkdown::render("/local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly/.snakemake/scripts/tmpy61rjv1e.align_bxstats.Rmd", output_file="/local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly/Align/bwa/reports/barcodes.summary.html", quiet=TRUE, knit_root_dir = "/local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly", params = list(rmd="/local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly/.snakemake/scripts/tmpy61rjv1e.align_bxstats.Rmd"))'' returned non-zero exit status 1.
[Tue Oct  8 14:55:28 2024]
Error in rule barcode_report:
    jobid: 13
    input: Align/bwa/reports/data/bxstats/C3-flyS3-10477404-22NLWYLT3-L2_trimmed.bxstats.gz
    output: Align/bwa/reports/barcodes.summary.html
    conda-env: /local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly/.snakemake/conda/52de4b1d31cff955b1d021a7fa1b5edf_

[Tue Oct  8 14:56:00 2024]
Finished job 8.
1 of 3 steps (33%) done
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2024-10-08T134431.955750.snakemake.log

**Comment by pdimens** [comment]
Is there a log file for the report? The new version of harpy should be outputting logs for reports and deleting them if they are empty. The log should be:

"Align/bwa/logs/reports/bxstats.report.log"`

**Comment by pdimens** [comment]
Can you verify that the input file is present?
```
Align/bwa/reports/data/bxstats/C3-flyS3-10477404-22NLWYLT3-L2_trimmed.bxstats.gz
```

**Comment by prmunn** [comment]
Nope, there is no file named: Align/bwa/logs/reports/bxstats.report.log
Yes, Align/bwa/reports/data/bxstats/C3-flyS3-10477404-22NLWYLT3-L2_trimmed.bxstats.gz is present

**Comment by pdimens** [comment]
That's so strange. Do you have the ability to run this workflow from scratch on a single sample?

Something like:
```bash
harpy align bwa -g genome.fasta -o test_bwa samples/sample1.*.fq.gz
```

**Comment by prmunn** [comment]
Does this help?

Rscript --vanilla -e 'rmarkdown::render("/local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly/.snakemake/scripts/tmpy61rjv1e.align_bxstats.Rmd", output_file="/local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly/Align/bwa/reports/barcodes.summary.html", quiet=TRUE, knit_root_dir = "/local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly", params = list(rmd="/local/workdir/prm88/GenomicsInnovation/haplotagging/GI24017_10477404_pheniqs_run_fly/.snakemake/scripts/tmpy61rjv1e.align_bxstats.Rmd"))'
Error: pandoc version 1.12.3 or higher is required and was not found (see the help page ?rmarkdown::pandoc_available).
Execution halted

**Comment by pdimens** [comment]
YES! That's the problem.

**Comment by prmunn** [comment]
Sorry - it took me a couple of tries to figure out which bit of the log file was the command I needed to run...

**Comment by pdimens** [comment]
Try this:

Add this line to the R conda env in `.harpy_envs`:
```
conda-forge::pandoc
```

and rerun with `harpy resume`

**Comment by pdimens** [comment]
If that solves the issue, I'll add that line to the code and push a hotfix as `1.19.1`

**Comment by prmunn** [comment]
Which directory is the one I want to resume from? Usage: harpy resume [OPTIONS] DIRECTORY

**Comment by pdimens** [comment]
`Align/bwa`  <- the output directory of your intended run

**Comment by pdimens** [comment]
@prmunn did this get resolved?

**Comment by prmunn** [comment]
@pdimens  I worked around this by bypassing the reporting step. I haven't tested it again with the report included.

**Comment by pdimens** [comment]
Alrighty. I'll close this for now. Please reopen this issue if it comes up again!
